test/extended/node_tuning/node_tuning.go

```diff
@@ -0,0 +1,115 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-node-tuning] NTO should", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		ntoNamespace        = "openshift-cluster-node-tuning-operator"
+		oc                  = exutil.NewCLIWithoutNamespace("nto")
+		buildPruningBaseDir = exutil.FixturePath("testdata", "node_tuning")
+		ntoStalldFile       = filepath.Join(buildPruningBaseDir, "nto-stalld.yaml")
+		stalldCurrentPID    string
+	)
+
+	// OCP-66086 - [OCPBUGS-11150] Node Tuning Operator - NTO Prevent from stalld continually restarting
+	// author: liqcui@redhat.com
+	// OCP Bugs: https://issues.redhat.com/browse/OCPBUGS-11150
+
+	g.It("OCP-66086 NTO Prevent from stalld continually restarting [Slow]", func() {
+		e2e.Logf("get the first rhcos worker nodes as label node")
+		firstCoreOSWorkerNodes, err := exutil.GetFirstCoreOsWorkerNode(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if len(firstCoreOSWorkerNodes) == 0 {
+			g.Skip("no rhcos worker node was found - skipping test ...")
+		}
+		e2e.Logf("the firstCoreOSWorkerNodes is:%v", firstCoreOSWorkerNodes)
+
+		defer oc.AsAdmin().WithoutNamespace().Run("label").Args("node", firstCoreOSWorkerNodes, "node-role.kubernetes.io/worker-stalld-", "--overwrite").Execute()
+		defer oc.AsAdmin().WithoutNamespace().Run("delete").Args("tuned", "openshift-stalld", "-n", ntoNamespace, "--ignore-not-found").Execute()
+
+		e2e.Logf("label the first rhcos node with node-role.kubernetes.io/worker-stalld=")
+		err = oc.AsAdmin().WithoutNamespace().Run("label").Args("node", firstCoreOSWorkerNodes, "node-role.kubernetes.io/worker-stalld=", "--overwrite").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("create custom profile openshift-stalld")
+		err = oc.AsAdmin().WithoutNamespace().Run("apply").Args("-f", ntoStalldFile, "-n", ntoNamespace).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("assert if the tuned openshift-stalld created successfully")
+		tunedStdOut, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("tuned", "-n", ntoNamespace).Output()
+		e2e.Logf("current tuned status is:\n%s,", tunedStdOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(tunedStdOut).NotTo(o.BeEmpty())
+		o.Expect(tunedStdOut).To(o.ContainSubstring("openshift-stalld"))
+
+		// Assert if profile applied to label node with re-try
+		o.Eventually(func() bool {
+			appliedStatus, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", ntoNamespace, "profile", firstCoreOSWorkerNodes, `-ojsonpath='{.status.conditions[?(@.type=="Applied")].status}'`).Output()
+			if err != nil || strings.Contains(appliedStatus, "False") || strings.Contains(appliedStatus, "Unknown") {
+				e2e.Logf("failed to apply custom profile to nodes, the status is %s and %v, check again", appliedStatus, err)
+			}
+			return strings.Contains(appliedStatus, "True")
+		}, 5*time.Second, time.Second).Should(o.BeTrue())
```

Comment:
what's the timeout on this retry?

Reply:
[5*time.Second, time.Second] the max retry is 5 times. every second retry 1 time

Reply:
thanks